src/bio.c

```diff
@@ -527,109 +518,114 @@ int wolfSSL_BIO_write(WOLFSSL_BIO* bio, const void* data, int len)
-
-    #ifndef WOLFCRYPT_ONLY
-        if (bio->type == WOLFSSL_BIO_SSL) {
-            /* already got eof, again is error */
-            if (front->eof) {
-                ret = SSL_FATAL_ERROR;
-            }
-            else {
-                ret = wolfSSL_BIO_SSL_write(bio, data, len, front);
-            }
-            /* Rest of chain is taken care of inside call */
-            break;
-        }
-
-        if (bio->type == WOLFSSL_BIO_MD) {
-            if (bio->next != NULL) { /* data passing through MD BIO */
-                ret = wolfSSL_BIO_MD_write(bio, data, len);
+                if (ret >= 0) {
+                    /* change so that data is formatted buffer */
+                    retB64 = wolfSSL_BIO_BASE64_write(bio, data, (word32)len,
+                            (byte*)frmt, &frmtSz);
+                    data = frmt;
+                    len  = frmtSz;
+                }
+            #endif /* WOLFSSL_BASE64_ENCODE */
+                break;
             }
-        }
-    #endif /* WOLFCRYPT_ONLY */
-
-        if (bio->type == WOLFSSL_BIO_SOCKET) {
-            ret = wolfIO_Send(bio->num, (char*)data, len, 0);
+            case WOLFSSL_BIO_BIO: /* write bios */
+                ret = wolfSSL_BIO_BIO_write(bio, data, len);
+                break;
+            case WOLFSSL_BIO_MEMORY:
+                ret = wolfSSL_BIO_MEMORY_write(bio, data, len);
+                break;
+            case WOLFSSL_BIO_FILE:
+            #ifndef NO_FILESYSTEM
+                if (bio->ptr) {
+                    ret = (int)XFWRITE(data, 1, len, (XFILE)bio->ptr);
+                }
+                #if !defined(USE_WINDOWS_API) && !defined(NO_WOLFSSL_DIR) && \
+                    !defined(WOLFSSL_NUCLEUS) && !defined(WOLFSSL_NUCLEUS_1_2)
+                else {
+                    ret = (int)XWRITE(bio->num, data, len);
+                }
+                #endif
+            #endif /* !NO_FILESYSTEM */
+                break;
+            case WOLFSSL_BIO_SSL:
+            #ifndef WOLFCRYPT_ONLY
+                /* already got eof, again is error */
+                if (front->eof) {
+                    ret = WOLFSSL_FATAL_ERROR;
+                }
+                else {
+                    ret = wolfSSL_BIO_SSL_write(bio, data, len, front);
+                }
+                /* Rest of chain is taken care of inside call */
+                goto exit_chain;
+            #endif
+                break;
+            case WOLFSSL_BIO_MD:
+            #ifndef WOLFCRYPT_ONLY
+                if (bio->next != NULL) { /* data passing through MD BIO */
+                    ret = wolfSSL_BIO_MD_write(bio, data, len);
+                }
+            #endif
+                break;
+            case WOLFSSL_BIO_SOCKET:
+                ret = wolfIO_Send(bio->num, (char*)data, len, 0);
+                break;
+            } /* switch */
         }
 
         /* advance to the next bio in list */
         bio = bio->next;
     }
 
+exit_chain:
```

Comment:
I just noticed this is only used if `#ifndef WOLFCRYPT_ONLY`. WIll fix.

Reply:
I noticed that we quietly skip over not compiled in BIO's. It would be beneficial to at least log not compiled in BIO's or even error out when encountering not supported BIO's.